Handle rejected promises

billymcintosh opened the pull request:
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bugfix to handle rejected promises in case of spelling errors or APIs returning bad data. 


* **What is the current behavior?** (You can also link to an open issue here)
When the marketFetcher received a warning and tries to log to the console, it finds a misspelt variable and throws rejection


* **What is the new behavior (if this is a feature change)?**
Fixes the spelling mistake and adds a listener to `gekko.js` to catch any other unhandled promises.   This listener only logs to the console for now.


* **Other information**:
Happy for the unhandledRejection process listener to be dumped but it will be worthwhile going forward when most people catch up with the latest node versions.  In future node versions, the process will exit on unhandled rejections.

billymcintosh committed:
Fix spelling error in variable causing rejected promise

askmike commented on gekko.js:
```diff
@@ -59,3 +59,7 @@ pipeline({
 });
 
+process.on('unhandledRejection', (reason, p) => {
+  console.log(`Unhandled Rejection at: ${p}, Reason: ${reason}`);
```
can we use the the logger that comes with Gekko, and use the `error` method to log to stdout?

That logger can be found here: https://github.com/askmike/gekko/blob/develop/core/log.js

billymcintosh replied:
Yep, makes sense.

billymcintosh replied:
Updated and pushed changes.

billymcintosh committed:
Add unhandledRejection process listener

This will handle any promise rejections and allow execution to
continue without restarting the node process.
